Assets/Scripts/HUD.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class HUD : MonoBehaviour
+{
+    [SerializeField] List<Sprite> healthList;
+    [SerializeField] Image healthImage;
+    public PlayerControllerV2 player;
+
+    private void Start()
+    {
+        player = GameObject.FindGameObjectWithTag("player").GetComponent<PlayerControllerV2>();
+    }
+
+    private void Update()
+    {
+        healthImage.sprite = healthList[player.curHealth];
+    }
+}
```

Comment:
I think this class is good, it has what you need but maybe it would be better to drag and drop the player object in the Unity inspector that way you don't have to use Gameobject.FindGameObjectWithTag and simply use GetComponent instead. Not only is it it easier to read, but it makes more sense since the player is already public and visible in the inspector